vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java

```diff
@@ -251,18 +245,8 @@ private void initBaseElementsAndListeners() {
                 event -> fireEvent(
                         new OpenedChangeEvent(this, event.isUserOriginated())));
 
-        getElement().addEventListener(OPENED_CHANGED_EVENT,
-                event -> removeAutoAdded());
-    }
-
-    /**
-     * Removes the notification from its parent if it was added automatically.
-     */
-    private void removeAutoAdded() {
-        if (autoAddedToTheUi && !isOpened()) {
-            autoAddedToTheUi = false;
-            getElement().removeFromParent();
-        }
+        // Initialize auto add behavior
+        new OverlayAutoAddController<>(this);
```

Comment:
This change brings a bonus benefit: it fixes an issue where Notification required an extra server visit to remove the auto-added notification element after calling `close()` on the server.